src/dfx/assets/build.rs

```diff
@@ -144,13 +144,48 @@ fn write_archive_accessor(fn_name: &str, f: &mut File) {
     .unwrap();
 }
 
-fn get_git_hash() -> Option<String> {
-    match Command::new("git").arg("describe").arg("--dirty").output() {
-        Ok(output) if output.status.success() => {
-            Some(String::from_utf8_lossy(&output.stdout).to_string())
+/// Gets a git tag with the least number of revs between HEAD of current branch and the tag,
+/// and combines is with SHA of the HEAD commit. Example of expected output: `0.12.0-beta.1-b9ace030`
+fn get_git_hash() -> Result<String, std::io::Error> {
+    let mut latest_tag: (u128, String) = (u128::MAX, String::from("0"));
```

Comment:
I'd suggest not to use a tuple here, rather than having a `latest_tag` that holds both of the following:
```suggestion
    let mut latest_distance = i128::MAX;
    let mut latest_tag = String::from("0");
```

Reply:
✅